app/src/main/java/com/github/gotify/messages/ListMessageAdapter.java

```diff
@@ -123,9 +124,37 @@ public long getItemId(int position) {
         @BindView(R.id.message_delete)
         ImageButton delete;
 
+        private boolean preciseDate;
+        private OffsetDateTime dateTime;
+
         ViewHolder(final View view) {
             super(view);
             ButterKnife.bind(this, view);
+            preciseDate = false;
+            dateTime = null;
+        }
+
+        void switchPreciseDate() {
+            preciseDate = !preciseDate;
+            updateDate();
+        }
+
+        void setDateTime(OffsetDateTime dateTime) {
+            this.dateTime = dateTime;
+            preciseDate = false;
+            updateDate();
+        }
+
+        void updateDate() {
+            String text = "?";
+            if (dateTime != null) {
+                if (preciseDate) {
+                    text = dateTime.toString();
```

Comment:
```suggestion
                    text = dateTime.truncatedTo(ChronoUnit.SECONDS).toString();
```
I think we can truncate millisecords and nanoseconds otherwise the text is pretty long and not really readable.
![2020-02-26_20-15](https://user-images.githubusercontent.com/14895212/75378928-78b63080-58cc-11ea-98a3-096e210624ec.png)

Reply:
Done